pkg/lib/authn/identity/ldap/provider.go

```diff
@@ -100,3 +109,35 @@ func sortIdentities(is []*identity.LDAP) {
 		return is[i].CreatedAt.Before(is[j].CreatedAt)
 	})
 }
+
+func (p *Provider) CreateNormalizedIdentitySpecFromLDAPEntry(serverConfig *config.LDAPServerConfig, entry *ldap.Entry) (*identity.Spec, error) {
+	userIDAttributeName := serverConfig.UserIDAttributeName
+	userIDAttributeValue := entry.GetRawAttributeValue(userIDAttributeName)
+
+	claims := map[string]interface{}{}
+	if v := entry.GetAttributeValue(ldap.AttributeMail.Name); v != "" {
+		claims[string(model.ClaimEmail)] = v
+	}
+	if v := entry.GetAttributeValue(ldap.AttributeMobile.Name); v != "" {
+		claims[string(model.ClaimPhoneNumber)] = v
+	}
+	if v := entry.GetAttributeValue(ldap.AttributeUID.Name); v != "" {
+		claims[string(model.ClaimPreferredUsername)] = v
+	}
+
+	err := p.StandardAttributesNormalizer.Normalize(claims)
+	if err != nil {
+		claims = map[string]interface{}{}
+	}
```

Comment:
I am not sure what we would want to handle here. Since ldap attributes are not guarantee, e.g. arbitrary value can be put in `email`. Would we want to normalise the attribute one by one or by all-or-nothing policy?

Reply:
@louischan-oursky

Reply:
You return `err` in this case. By returning error, we will not store anything malformed into the database. And we also not silence the error. Currently, this line silences the error, which is super subtle.